From 66242aa76a5d59fb4659551575c1fbb42e50b8fb Mon Sep 17 00:00:00 2001
From: Giovanni Campagna <gcampagna@src.gnome.org>
Date: Tue, 18 Oct 2011 10:19:50 +0200
Subject: [PATCH 1/2] Make dock extension actually work with gnome-3.2

Without this patch dock is invisible, but reacts on mouse clicks

Signed-off-by: Vasily Khoruzhick <anarsoul@gmail.com>
---
 extensions/dock/extension.js |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/extensions/dock/extension.js b/extensions/dock/extension.js
index 240f8a8..02c9650 100644
--- a/extensions/dock/extension.js
+++ b/extensions/dock/extension.js
@@ -304,7 +304,6 @@ Dock.prototype = {
             this.actor.show();
         }));
         Main.layoutManager.addChrome(this.actor);
-        this.actor.lower_bottom();
 
         //hidden
         this._settings.connect('changed::'+DOCK_POSITION_KEY, Lang.bind(this, function (){
@@ -372,6 +371,8 @@ Dock.prototype = {
 
         this.actor.connect('leave-event', Lang.bind(this, this._hideDock));
         this.actor.connect('enter-event', Lang.bind(this, this._showDock));
+
+        this._hideDock();
     },
 
     destroy: function() {
@@ -861,4 +862,4 @@ function enable() {
 function disable() {
     dock.destroy();
     dock = null;
-}
\ No newline at end of file
+}
-- 
1.7.7.2

